sdks/nodejs/src/agonesSDK.d.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Alpha from './alpha';
+
+type Seconds = number;
+
+export declare class AgonesSDK {
+    constructor();
+
+    alpha: Alpha
+
+    get port(): string
+
+    connect(): Promise<void>
+
+    close(): void
+
+    ready(): Promise<Record<string, any>>
```

Comment:
When using TS I usually use `Record<string, unknown>` here as would cast the type to something useful but maybe `any` is more flexible?
We could also supply full return types, e.g. interfaces/types for all these, or better to do in a separate PR?

Reply:
As for `unknown` and `any`, `unknown` is definitely better when you can distinguish type, there's no `instanceof` type check you can do, or another field like `isAxiosError` to distinguish a particular type. The problem I see with `unknown` is that in this particular case, you have to write
```
if(typeof awaitedValue === 'object') {
  if(typeof awaitedValue.prop1 === 'object') {
    awaitedValue.prop1.prop1A
  }
}
```
with `any` you can just use awaitedValue?prop1?prop1A with the same security.
Maybe I am missing something?

Reply:
Can we provide full return types? That would be much better. I just started working with Agones last week, but for me the real result only returns for `getGameServer` and `watchGameServer` methods, for the rest sdk returns an empty object, maybe something is wrong in my cluster.
if we could specify the exact types, I would definitely do in this PR